File: qtile/.config/qtile/ui/bar.py
from libqtile import bar, qtile

from core.helper import load_module
from options import default_bar_type, default_colorscheme, default_options

# dynamic module loading
colorscheme_module_path = f"themes.colorschemes.{default_colorscheme}.colors"
colors = load_module(colorscheme_module_path)

widgets_module_path = f"widgets.{default_bar_type}.widgets"
w = load_module(widgets_module_path)

custom_widgets_module_path = f"widgets.{default_bar_type}.custom_widgets"
cw = load_module(custom_widgets_module_path)
# ---------------------

if qtile.core.name == "x11":
    widget_list = (
        w.app_launcher
        + w.separator_small
        + w.tray
        + w.separator_small
        + w.windowname
        + w.separator_auto
        + w.client_count
        + w.separator_small
        + w.groups
        + w.separator_small
        + w.layout_icon
        + w.separator_auto
        + w.widgetbox_info
        + w.separator_small
        + w.battery
        + w.separator_small
        + w.time
        + w.separator_small
        + w.date
    )
else:
    widget_list = (
        w.app_launcher
        + w.windowname
        + w.separator_auto
        + w.client_count
        + w.separator_small
        + w.groups
        + w.separator_small
        + w.layout_icon
        + w.separator_auto
        + w.widgetbox_info
        + w.separator_small
        + w.battery
        + w.separator_small
        + w.time
        + w.separator_small
        + w.date
    )

gaps_size = default_options["gaps_size"]

top_bar = bar.Bar(
    widget_list,
    size=30,
    background=colors.bar_colors["bg"],
    margin=[gaps_size, gaps_size, 0, gaps_size],  # [top, right, bottom, left]
    border_width=[5, 5, 5, 5],  # [top, right, bottom, left]
    border_color=[
        colors.bar_colors["border"],
        colors.bar_colors["border"],
        colors.bar_colors["border"],
        colors.bar_colors["border"],
    ],
)

File: stow_packages/awesome/.config/awesome/visual/bar.lua
local awful = require("awful")
local beautiful = require("beautiful")
local wibox = require("wibox")
local gears = require("gears")

require("widgets")
require("core.utils")

require("visual.options")
require("visual.themes." .. THEME_COLORSCHEME .. ".colors")

local systray = require("widgets.tray")
local c_count_table = require("widgets.client_count")

-- {{{ Wibar
screen.connect_signal(
  "request::desktop_decoration",
  function(s)
    -- Each screen has its own tag table.
    --[[
    awful.tag(
      { "", "", "", "", "", "󰚢", "󰸳", "", "󰆩" },
      -- 󰾔
      s,
      awful.layout.layouts[1]
    )
    ]]
    --

    --
    awful.tag.add("", {
      index = 1,
      screen = s,
      layout = awful.layout.suit.tile,
      selected = true,
    })
    --󱃖
    awful.tag.add("", {
      index = 2,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --
    awful.tag.add("", {
      index = 3,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --
    awful.tag.add("", {
      index = 4,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --
    awful.tag.add("", {
      index = 5,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --
    awful.tag.add("", {
      index = 6,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --󰚢
    awful.tag.add("", {
      index = 7,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --󰸳
    awful.tag.add("", {
      index = 8,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })
    --
    awful.tag.add("", {
      index = 9,
      screen = s,
      layout = awful.layout.suit.max,
      --selected = true,
    })

    -- Create a promptbox for each screen
    s.mypromptbox = awful.widget.prompt()

    s.layoutbox = get_layout_indicator(s)

    s.taglist = get_taglist(s)

    local client_count_widget = c_count_table.client_count_widget

    -- s.tasklist = get_tasklist(s)

    -- s.systray = get_systray(s)

    -- main bar
    -- Create the wibox
    s.mywibox = awful.wibar {
      widget = {
        { -- Left widgets
          {
            -- mylauncher,
            app_launcher,
            -- s.tasklist,
            s.mypromptbox,
            separator,

            spacing = 10,
            layout = wibox.layout.fixed.horizontal,
          },
          halign = "left",
          layout = wibox.container.place,
        },
        { -- Middle widgets
          {
            client_count_widget,
            s.taglist,
            s.layoutbox,

            spacing = 10,
            layout = wibox.layout.fixed.horizontal,
          },
          halign = "center",
          layout = wibox.container.place,
        },
        { -- Right widgets
          {
            --keyboard_layout,
            battery,
            time,
            date,
            systray.tray,

            spacing = 10,
            layout = wibox.layout.fixed.horizontal,
          },
          halign = "right",
          layout = wibox.container.place,
        },
        expand = "none",
        layout = wibox.layout.align.horizontal,
      },
      --screen = s,
      bg = bar_colors.bg,
      position = "top",
      height = 35,
      border_width = 6,
      border_color = bar_colors.border,
      margins = {
        --top = beautiful.useless_gap + beautiful.border_width,
        left = beautiful.useless_gap + (beautiful.border_width * 2),
        right = beautiful.useless_gap + (beautiful.border_width * 2),
      },
      shape = function(cr, w, h)
        gears.shape.partially_rounded_rect(cr, w, h, false, false, true, true, 15) -- t-left, t-right, b-right, b-left
      end,
    }
  end
)
-- }}}
